### ISSUE TITLE
The curator dashboard is not working correctly on PROD

### ISSUE THREAD

**Issue by bwbaker1** [NASA-IMPACT/nasa-apt#643]
@wrynearson @deborahUAH 

### **Description**
The curator dashboard is not working properly on PROD.  Documents are not loading and an error message is given.
Note: The curator dashboard IS working correctly on staging.

### **Tasks:**
- [ ] Determine what is causing the dashboard not to work
- [ ] Implement a fix
- [ ] Test to ensure the dashboard is working correctly

### **Acceptance Criteria:**
- [ ] Curator dashboard is working properly including all documents appearing
- [ ] The dashboard has been tested and is working

#### Example

![Image](https://user-images.githubusercontent.com/56319064/220739729-29051f15-4f46-4658-bdc8-714df554edfc.png)

**Comment by wrynearson** [comment]
cc @sunu @batpad

**Comment by wrynearson** [comment]
I think this issue is not that the dashboard isn't loading any documents (unless you're sure that you're the lead author on several documents), but instead that we're displaying multiple dashboards for curators.

![image](https://user-images.githubusercontent.com/25919630/220840964-404cb1f9-660b-4d82-adbf-3437019ee8c6.png)

We had a similar issue on SNWG for curators, which was solved with this PR: https://github.com/NASA-IMPACT/snwg-frontend/pull/103

@frozenhelium we should port this over to APT. I'll put this in the next sprint (sprint 4). cc @batpad @sunu

**Comment by deborahUAH** [comment]
@bwbaker1 @wrynearson @frozenhelium 
The curator dashboard worked until just recently.  I was always able to see the ATBDs my account was lead author or contributor for (top half which still works) AND I could see the bottom list of all ATBDs (needed as curator).  

Something must have just changed to create this error.  The message has been consistent all this week.
Pleas note, we can not test for needed notifications until curators can again see other ATBD documents in the lower list.

**Comment by wrynearson** [comment]
@deborahUAH is the desired behavoir to have two dashboards, as we currently have for curators on production? I think this is actually a bug. Right now, we're showing the curator dashboard _and_ the contributor dashboard. Contributors can only see the contributor dashboard. If it's the desired behavior, then we can keep it.

@thenav56 @sunu @frozenhelium – could one of you look into this, if you're able to (it's on production in MCP). I don't recall making any pushes to production recently, so I don't know what could have caused this change.

**Comment by deborahUAH** [comment]
@wrynearson the curator should have two dashboards (I am a curator, as is Brad). On the curator dashboard, we should see the docs we are part of on the top list (I have a few I am "author" on) and we should also see the bottom list of all documents for curation.

**Comment by deborahUAH** [comment]
fyi, I just tested production again and the bottom list is now showing.  The staging has the OTP active now so I did not get in to test - need the code.

**Comment by wrynearson** [comment]
OK, if you need the OTP for logging into a dummy account, please let me know, or someone from DevSeed. I can help you set it up for your own account if you'd like.

**Comment by wrynearson** [comment]
@frozenhelium or @sunu – do you have any idea how this could have broken? It's working on staging and production now.

**Comment by sunu** [comment]
@wrynearson from the screenshot above it looks like the backend was returning a 500 error. But without access to api logs, it's hard to tell why

**Comment by wrynearson** [comment]
OK thanks @sunu. I'll close this ticket for now but if the issue returns, we can look into it.

**Comment by wrynearson** [comment]
Hey @frozenhelium or @sunu, the bug is back on production. It looks like a backend issue.

Since it's in MCP, I'm tagging @naomatheus to see if he can have a look. This is a pretty big bug, so quick action would be appreciated.

**Comment by wrynearson** [comment]
@leothomas, @naomatheus and I looked into the issue on Feb 28th. Thanks @leothomas for summarizing the following! 👇 

## Cause of issue
The issue is caused by a "ghost user" (a user that is a contributing author on report 84 but is not present in Cognito).

The dashboard fails due to the following process:
1. Curator loads documents
2. Backend goes through all documents, finds the cognito subs (uuids) of each user associated with the document (owner, author, reviewer, created_by, last_updated_by)
3. Backend queries cognito and matches the subs (uuids) from the documents with the subs found in cognito
    - :green_circle: Backend finds all users in cognito, replaces subs with user information, appropriate for the user making the request (eg: curator can see all info, author can't see reviewer names, etc) 
    - :red_circle: **One of the cognito subs from the document ISN'T found in Cognito --> Error 500** (cause of this bug)

To summarize, the error is that one document in the prod database has a Cognito sub in the author field **which does not exist in Cognito.**

## Further context
- Report 84 is blank but malformed (missing default fields)
- The author (@cw0045) shared that nothing strange happened in the creation process, and that she didn't add any users as report collaborators.

_tl;dr – we're not sure how this situation happened, but we know what is causing the error._

## Next steps
### Immediate: 
- [x] Replace the missing sub in the authors field with the sub in the owner field: 
    - We have verified that sub exists in cognito. This is preferable to deleting the document, which may orphan some resources elsewhere in the database or other resources (opensearch, S3 documents, etc)
    - Note: if the document failed to create and is in some sort of intermediary state, the document may still fail to load
### Long term: 
- [ ] Update error handling such that any one document failing to load wouldn't cause the entire request to fail

cc @batpad @deborahUAH @bwbaker1

**Comment by bwbaker1** [comment]
@wrynearson Glad this happened now and not further on down the road.  In terms of getting the dashboard functioning properly again, is this a quick fix?

**Comment by wrynearson** [comment]
@bwbaker1 agreed. It should be a quick fix to implement the "immediate" fix. We'll look at that in a few hours. It involves making a manual database change, so we want to be cautious and deliberate before fixing it. If everything goes as planned, it will be fixed later today.

**Comment by wrynearson** [comment]
This is fixed on production with the "immediate fix":
> Replace the missing sub in the authors field with the sub in the owner field:

We removed the ghost author from author array, then replace that UUID in the ATBD and ATBD version with owner’s UUID (including the created by and last modified by fields).

For the record, [this](https://www.earthdata.nasa.gov/apt/documents/untitled-document-21/v1.0) was the document causing the issue, which is now viewable and has no further issues

Next steps are to ticket out the "long term" fix, which I will do. I will close this ticket, but if this issue reappears, please let me know and we can reopen it.

cc @bwbaker1 @deborahUAH @batpad
